src/coreclr/jit/importer.cpp

```diff
@@ -9497,15 +9495,67 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                                 if (info.compCompHnd->getReadonlyStaticFieldValue(resolvedToken.hField, buffer,
                                                                                   genTypeSize(lclTyp)))
                                 {
-                                    GenTree* cnsValue = impImportStaticReadOnlyField(buffer, bufferSize, lclTyp);
+                                    GenTree* cnsValue = impImportStaticReadOnlyField(buffer, lclTyp);
                                     if (cnsValue != nullptr)
                                     {
                                         op1 = cnsValue;
                                         goto FIELD_DONE;
                                     }
                                 }
                             }
+                            else if (lclTyp == TYP_STRUCT)
```

Comment:
Can we pull all this out into the `impImportStaticReadOnlyField` helper method?

Reply:
Moved to a separate function